src/main/java/mate/academy/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,68 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
```

Comment:
As a suggestion, to update user
```suggestion
            session.refresh(shoppingCart.getUser());
            session.persist(shoppingCart);
```